for_dict_challenges_bonus.py

```diff
@@ -55,5 +56,79 @@ def generate_chat_history():
     return messages
 
 
+import pprint
+from collections import Counter
+def task1(messages):  # 1. Вывести айди пользователя, который написал больше всех сообщений.
+    sent_by = [each['sent_by'] for each in messages]
+    print(f"\nMost messages from user {Counter(sent_by).most_common()[0][0]}")
+
+def task2(messages):  # 2. Вывести айди пользователя, на сообщения которого больше всего отвечали.
+    reply_for = [str(each['reply_for']) for each in messages  # генерирую список сообщений,
+                 # на которые отвечали
+                 if each['reply_for'] is not None]
+    most_replied_user = [each['sent_by'] for each in messages
+                         if str(each['id']) == Counter(reply_for)
+                         .most_common()[0][0]]
+    print(f"\nThe most replyable user is {most_replied_user[0]}")
+
+def task3(messages):  # 3. Вывести айди пользователей, сообщения которых видело больше всего
+    # уникальных пользователей.
+    unique_users = {}
+    for every_user in [each['sent_by'] for each in messages]:
+        for seen_list in [each_seen['seen_by'] for each_seen in messages
+                          if each_seen['sent_by'] == every_user]:
+            if every_user in unique_users:
+                unique_users[every_user].update(seen_list)
+            else:
+                unique_users[every_user] = set(seen_list)
+    print('\nThree users with most unique views')
+    for counter, each in enumerate(Counter(unique_users).most_common(), start=1):
+        print(f"User {each[0]} has viewrs {each[1]}")
+        if counter == 3: break
+
+
+def task4(messages):  # 4. Определить, когда в чате больше всего сообщений: утром (до 12 часов),
+# днём (12-18 часов) или вечером (после 18 часов).
+    rush_hours = {'Morning': 0, 'Daytime': 0, 'Evening': 0}
+    for each in messages:
+        if each['sent_at'].hour < 12:
+            rush_hours['Morning'] += 1
+        elif each['sent_at'].hour < 18:
+            rush_hours['Daytime'] += 1
+        else:
+            rush_hours['Evening'] += 1
+    print(f"\nThe rush hours is {Counter(rush_hours).most_common()[0][0]}")
```

Comment:
Ну 3 ключа всего же, ну зачем Counter тащить

Reply:
```suggestion
          f"{[x for x in rush_hours if rush_hours[x] == max(rush_hours.values())][0]}")
```
Кроме такого варианта ничего не приходит в голову. Так ок?

Reply:
Да можно даже три отдельных переменный и потом максимум найти.